### ISSUE TITLE
acc on lfw cannot be imporoved over 0.9

### ISSUE THREAD

**Issue by Carbord** [Joker316701882/Additive-Margin-Softmax#20]
Hello, I have tried several combinations of parameters while running train.py, such as epoch_size =200, max_epoch_num =500, keep_probability = 0.9. And the weight_decay and batch_size  are set to 5e-4 and 256 as suggested. But the accuracy is around 0.9 on the lfw dataset.  So  I hope for some advice from you by setting training parameters . 
What'more, I wonder whether it is because  that I have not used any per-trained model. I'm gratitude to wait for your reply soon.

**Comment by Joker316701882** [comment]
@Carbord 
Well it shouldn't be if you follow all instructions in README.md. Perhaps you didn't align lfw, or you didn't train enough time.

**Comment by Carbord** [comment]
> @Carbord
> Well it shouldn't be if you follow all instructions in README.md. Perhaps you didn't align lfw, or you didn't train enough time.

As is mentioned is the README.md, I should align both the training set and lfw. Is this mean that I should use the align_lfw.py for twice? My setting is like following: 
Step1 --"python align_lfw.py --input-dir /DataSet/lfwdataset/lfw-deepfunneled --output-dir /home/myname/lfwdataset"
Step2  --"python train.py --data_dir /home/myname/lfwdataset  --random_flip --learning_rate -1 --learning_rate_schedule_file ./data/learning_rate_AM_softmax.txt --lfw_dir /home/myrname/lfwdataset --keep_probability 0.8 --weight_decay 5e-4"
Is there any problem? Hope for your reply soon, thank you~

**Comment by Carbord** [comment]
As is mentioned is the README.md, I should align both the training set and lfw. Is this mean that I should use the align_lfw.py for twice? My setting is like following:
Step1 --"python align_lfw.py --input-dir /DataSet/lfwdataset/lfw-deepfunneled --output-dir /home/myname/lfwdataset"
Step2 --"python train.py --data_dir /home/myname/lfwdataset --random_flip --learning_rate -1 --learning_rate_schedule_file ./data/learning_rate_AM_softmax.txt --lfw_dir /home/myrname/lfwdataset --keep_probability 0.8 --weight_decay 5e-4"
Is there any problems? Hope for your reply soon, thank you~

**Comment by wxl3d** [comment]
hi, i want to know which algorithm used to align lfw. i just use the mctnn to crop the train sample and lfw without other operation. But i can't get the high acc with additive margin softmax loss. Wait for your advice urgently! Many thanks!

**Comment by Carbord** [comment]
> hi, i want to know which algorithm used to align lfw. i just use the mctnn to crop the train sample and lfw without other operation. But i can't get the high acc with additive margin softmax loss. Wait for your advice urgently! Many thanks!

I just follow all instructions in README.md, thus I use align_lfw.py to align my dataset.
